src/libs/metadata_view.c

```diff
@@ -791,46 +910,306 @@ void connect_key_accels(dt_lib_module_t *self)
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, -1, DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_box_pack_start(GTK_BOX(area), w, TRUE, TRUE, 0);
+
+  GtkListStore *store = gtk_list_store_new(DT_METADATA_PREF_NUM_COLS,
+                                           G_TYPE_INT, G_TYPE_STRING, G_TYPE_BOOLEAN);
+  GtkTreeModel *model = GTK_TREE_MODEL(store);
 
-  /* initialize the metadata name/value labels */
-  for(int k = 0; k < md_size; k++)
   {
-    d->name[k] = GTK_LABEL(gtk_label_new(_md_labels[k]));;
-    gtk_widget_set_halign(GTK_WIDGET(d->name[k]), GTK_ALIGN_START);
-    gtk_label_set_xalign (d->name[k], 0.0f);
-    gtk_label_set_ellipsize(d->name[k], PANGO_ELLIPSIZE_END);
-    gtk_widget_set_tooltip_text(GTK_WIDGET(d->name[k]), _md_labels[k]);
-
-    d->metadata[k] = GTK_LABEL(gtk_label_new("-"));
-    gtk_widget_set_name(GTK_WIDGET(d->metadata[k]), "brightbg");
-    gtk_label_set_selectable(d->metadata[k], TRUE);
-    gtk_widget_set_halign(GTK_WIDGET(d->metadata[k]), GTK_ALIGN_FILL);
-    gtk_label_set_xalign (d->metadata[k], 0.0f);
-    gtk_label_set_ellipsize(GTK_LABEL(d->metadata[k]),
-                            k == md_exif_model || k == md_exif_lens || k == md_exif_maker
-                            ? PANGO_ELLIPSIZE_END : PANGO_ELLIPSIZE_MIDDLE);
-    if(k == md_internal_filmroll)
+    GtkTreeIter sort_iter;
+    gboolean valid = gtk_tree_model_get_iter_first(d->sort_model, &sort_iter);
+    while(valid)
     {
-      // film roll jump to:
-      g_signal_connect(G_OBJECT(GTK_WIDGET(d->metadata[k])), "button-press-event", G_CALLBACK(_filmroll_clicked), NULL);
+      GtkTreeIter iter;
+      int index;
+      gboolean visible;
+      char *name = NULL;
+      gtk_tree_model_get(d->sort_model, &sort_iter,
+                         DT_METADATA_COL_INDEX, &index,
+                         DT_METADATA_COL_NAME_L, &name,
+                         DT_METADATA_COL_VISIBLE, &visible,
+                         -1);
+      if(!_is_metadata_ui(index))
+        continue;
+      gtk_list_store_append(store, &iter);
+      gtk_list_store_set(store, &iter,
+                         DT_METADATA_PREF_COL_INDEX, index,
+                         DT_METADATA_PREF_COL_NAME_L, name,
+                         DT_METADATA_PREF_COL_VISIBLE, visible,
+                         -1);
+      g_free(name);
+      valid = gtk_tree_model_iter_next(d->sort_model, &sort_iter);
     }
+  }
+
+  GtkWidget *view = gtk_tree_view_new_with_model(model);
+  g_object_unref(model);
+  GtkCellRenderer *renderer = gtk_cell_renderer_text_new();
+  GtkTreeViewColumn *column = gtk_tree_view_column_new_with_attributes(_("metadata"), renderer,
+                                                    "text", DT_METADATA_PREF_COL_NAME_L, NULL);
```

Comment:
```
  g_object_set(renderer, "ellipsize", PANGO_ELLIPSIZE_END, NULL);
  gtk_tree_view_column_set_expand(column, TRUE);
```

Reply:
Great. This works.
I had already try gtk_tree_view_column_set_expand() alone but this did not work.